### PR TITLE
Add Simpread extension

### PR DESCRIPTION
**Pull request by Kenshin**
## Description

Search and open [SimpRead Unreader](https://simpread.pro)

## Type of change

- New extension

## Screencast

https://user-images.githubusercontent.com/81074/143815314-e6270ced-93e7-4e14-a708-1ce26ba756e1.mp4

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I checked that extension's icon has size of `512x512`

### CODE REVIEW

**Review by pernielsentikaer — COMMENTED** [review]
Hi @Kenshin 👋 

Thanks for the PR, it looks fantastic 🔥 

Can you maybe record a screencast of the flow since I don't have a way to test it locally, that would help my review a lot 😄

**Comment by Kenshin** [comment]
@pernielsentikaer 

Hi,  Video is here

https://user-images.githubusercontent.com/81074/143815314-e6270ced-93e7-4e14-a708-1ce26ba756e1.mp4

and test json file

1. plz download this [simpread_config.json.zip](https://github.com/raycast/extensions/files/7624068/simpread_config.json.zip) and unzip. e.g. `/Users/xxx/Downloads/simpread_config.json` 

2. set this path to Welcome 



3. <kbd>sr</kbd>  like it's  👇 



Done

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @Kenshin 👋 

Thanks for the extension 🔥 , I have a few comments 📄 

- What is simpRead excatly? (to make a better (sub)title) 
- I suggest removing the “SimpRead Config Setting” since it's already possible in settings
- There is [empty state flicker](https://developers.raycast.com/basics/prepare-an-extension-for-store#empty-states) upon opening, can you look into this?
- You could store the default favicon in assets folder

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by Kenshin** [comment]
Hi @pernielsentikaer

I update some new soure.

> What is simpRead excatly? (to make a better (sub)title)

Current title is **SimpRead** and description like is snapshot  👇 

image

iShot2021-12-02 16 32 12

Next version **add url to simpred**, so title is simpread like github

image

> I suggest removing the “SimpRead Config Setting” since it's already possible in settings

When removed it, user can't be re-set again.  😂 

***

And others is Done, source is here https://github.com/raycast/extensions/pull/435/commits/6bc50e3f335b58a39c2874d60e2e682f5caac059  😁 

- [x] There is empty state flicker upon opening, can you look into this?
- [x] You could store the default favicon in assets folder

File: extensions/simpread/package.json [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    {
+      "name": "src",
+      "title": "SimpRead Config Setting",
+      "description": "Set local simpread_config.json path",
+      "mode": "view"
+    },
+    {
+      "name": "sr",
+      "title": "SimpRead Unreader",
+      "description": "List of SimpRead Unreader",
```
Comment (pernielsentikaer) [inline extensions/simpread/package.json:19]:
```suggestion
      "title": "Browse Posts",
      "subtitle": "SimpRead",
      "description": "Browse SimpRead Posts",
```
Something like above

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @Kenshin 👋 

It's still possible to set another path within Raycast, see below:
aryygUufTuRzarWb

I was thinking about setting it like suggested if it match

**Comment by Kenshin** [comment]
Hi @pernielsentikaer

All is Done https://github.com/raycast/extensions/pull/435/commits/82e34fbb0d14aed9effd70ea34af673fd76d5061

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @Kenshin 🔥 

Thanks for your contribution, I'm going to accept it and then there will be a final review before shipping 🚢 

Have a nice day 🎄

**Comment by fe9lix** [comment]
@Kenshin  @pernielsentikaer Is this ready to be merged? Some remarks: 
- Please use standard code formatting via Prettier
- There are a couple of typos in the README

**Comment by Kenshin** [comment]
@fe9lix Hi,  3ks reviews

> There are a couple of typos in the README

What exactly is the error?

> Please use standard code formatting via Prettier

Not every merged project uses Prettier  😂

**Comment by fe9lix** [comment]
@Kenshin I've left some inline comments re the typos. As for the formatting: that might be the case, but it's still preferable to have consistent formatting as far as possible; this is why we've added Prettier as default and it will make future changes easier since another author wouldn't need to reformat all of you code (leading to unnecessary diffs in the review).

**Comment by Kenshin** [comment]
@fe9lix 

Hi, I added `prettier` to this project. https://github.com/raycast/extensions/pull/435/commits/c19472aab054709bae6e5dccff3a1d1f03287db9

File: extensions/simpread/src/sr.tsx [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+  const turndown = new TurndownService(),
+    fmtDate = (date: Date) => {
+      const format = (value: any) => (value = value < 10 ? '0' + value : value);
+      return (
+        date.getFullYear() +
+        '年' +
+        format(date.getMonth() + 1) +
+        '月' +
+        format(date.getDate()) +
+        '日 ' +
```
Comment (pernielsentikaer) [inline extensions/simpread/src/sr.tsx:63]: At the moment, Raycast doesn't support localization and only supports US English. Therefore, please avoid introducing your custom way to localize your extension

File: extensions/simpread/src/sr.tsx [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+***
+`;
+    });
+  const host = new URL(unread.url).hostname,
+    desc = unread.desc ? `> ${unread.desc}` : '',
+    note = unread.note ? `> ${unread.note}` : '',
+    template = `
+# ${unread.title}
+
+> 原文地址 [${host}](${unread.url}) at ${unread.create}
```
Comment (pernielsentikaer) [inline extensions/simpread/src/sr.tsx:106]: At the moment, Raycast doesn't support localization and only supports US English. Therefore, please avoid introducing your custom way to localize your extension

File: extensions/simpread/src/sr.tsx [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+        setState({ items: items });
+      } catch (error) {
+        setState({ error: new Error('Something went wrong') });
+      }
+    }
+    fetchUnrdist();
+  }, []);
+
+  if (state.error) {
+    showToast(ToastStyle.Failure, '获取失败，请确认是否存在 simpread_config.json');
```
Comment (pernielsentikaer) [inline extensions/simpread/src/sr.tsx:186]: At the moment, Raycast doesn't support localization and only supports US English. Therefore, please avoid introducing your custom way to localize your extension

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Tried to correct some spelling errors in the README, please check and correct the one making sense – also check the other correction about localization.

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by Kenshin** [comment]
@pernielsentikaer

Hi, I fixed spelling errors and changed localize. https://github.com/raycast/extensions/pull/435/commits/45080907e1ae764763f34de3fc9af9d04233031a

🙏

**Comment by Kenshin** [comment]
@pernielsentikaer

Hi, I fixed spelling errors and changed localize.https://github.com/raycast/extensions/pull/435/commits/45080907e1ae764763f34de3fc9af9d04233031a

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @Kenshin 👋 

Looks like we're close now 💪 The format of the date is not correctly, could you do something like “2021/11-25”

CzVB1RnM4W2WYNjN

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by Kenshin** [comment]
@pernielsentikaer

> The format of the date is not correctly, could you do something like “2021/11-25”

Hi, Fixed it. https://github.com/raycast/extensions/pull/435/commits/30158ee0367bbe8e30919f577e27d1f643269053

image

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @Kenshin 👋 

It was actually the date in the bottom that you're formatting I was pointing at, the one in top is from the received data and are allowed since it's from the content. 

ikmWNxA8pNSdZB4l

Sorry for the confusing 😄 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by Kenshin** [comment]
@pernielsentikaer

Sorry, It was my carelessness that wasted your review time, It's okay now. https://github.com/raycast/extensions/pull/435/commits/1147890d6b9d0572e0dc55318ad86ba622f933bb

image

**Comment by Kenshin** [comment]
@pernielsentikaer

Sorry, It was my carelessness that wasted your review time, It's okay now. https://github.com/raycast/extensions/pull/435/commits/1147890d6b9d0572e0dc55318ad86ba622f933bb

image

**Comment by Kenshin** [comment]
@pernielsentikaer

Sorry, It was my carelessness that wasted your review time, It's okay now. https://github.com/raycast/extensions/pull/435/commits/1147890d6b9d0572e0dc55318ad86ba622f933bb

image

**Review by pernielsentikaer — APPROVED** [review]
Hi @Kenshin 👋 

I'm going to accept it and then will my colleague do a final review before it's finally merged 💎

**Comment by Kenshin** [comment]
@pernielsentikaer

Haha, it's been too long, but it's a good thing it's finally done. 👋

**Comment by Kenshin** [comment]
@pernielsentikaer @fe9lix 👋

Thanks published and reviews 🙏 This is my New year's gift.

**Pull request merged**